src/api/order-v4.ts

```diff
@@ -0,0 +1,75 @@
+// add requestId
+import type { VercelRequest, VercelResponse } from '@vercel/node'
+import { decodeJWT } from '../auth'
+import { fulfillmentService, inventoryService, paymentService } from '../services'
+
+async function retry<Result>(
+  serviceCall: () => Promise<Result>,
+  maxAttempts = 10,
+  callTimeout = 30 * 1000,
+  initialInterval = 1000
+): Promise<Result> {
+  let result: Result | undefined
+  let error: any
+  let attempt = 0
+  while (attempt < maxAttempts) {
+    try {
+      result = await Promise.race([
+        serviceCall(),
+        new Promise<undefined>((resolve) => setTimeout(resolve, callTimeout, undefined)),
+      ])
+      const timedOut = result === undefined
+      if (timedOut) {
+        throw new Error('Timed out')
+      }
+      break
+    } catch (e) {
+      error = e
+      await new Promise((resolve) => setTimeout(resolve, initialInterval * Math.pow(2, attempt++)))
```

Comment:
You could use https://nodejs.org/api/timers.html#timerspromisessettimeoutdelay-value-options if assuming node >= 16